doc/latex.rst

```diff
@@ -1811,12 +1811,12 @@ Miscellany
 .. hint::
 
    As an experimental feature, Sphinx can use user-defined template file for
-   LaTeX source if you have a file named ``_templates/latex.tex_t`` in your
+   LaTeX source if you have a file named ``_templates/latex.tex.jinja`` in your
    project.
 
-   Additional files ``longtable.tex_t``, ``tabulary.tex_t`` and
-   ``tabular.tex_t`` can be added to ``_templates/`` to configure some aspects
-   of table rendering (such as the caption position).
+   Additional files ``longtable.tex.jinja``, ``tabulary.tex.jinja`` and
+   ``tabular.tex.jinja`` can be added to ``_templates/`` to configure some
+   aspects of table rendering (such as the caption position).
```

Comment:
I think it's worth extending this documentation to mention the [non-default syntax](https://github.com/sphinx-doc/sphinx/blob/9299003d40c775150c53366dcfdd4d944d0da1af/sphinx/util/template.py#L77-L84) for the Jinja2 files in this context.

Reply:
Relates-to: #9173

Reply:
(I'll try to find time to offer some updated documentation related to this in a separate pull request)

Reply:
> I think it's worth extending this documentation to mention the [non-default syntax](https://github.com/sphinx-doc/sphinx/blob/9299003d40c775150c53366dcfdd4d944d0da1af/sphinx/util/template.py#L77-L84) for the Jinja2 files in this context.

Thanks for all your hard work @jayaddison.  Perhaps we can document the [non-default syntax](https://github.com/sphinx-doc/sphinx/blob/9299003d40c775150c53366dcfdd4d944d0da1af/sphinx/util/template.py#L77-L84) as you rightly insisted upon  as a comment inside the template files themselves?  People intending to replace them by custom templates will probably start at looking how the originals are done.  Perhaps
```diff
diff --git a/sphinx/templates/latex/latex.tex_t b/sphinx/templates/latex/latex.tex_t
index deb030504..aae912a09 100644
--- a/sphinx/templates/latex/latex.tex_t
+++ b/sphinx/templates/latex/latex.tex_t
@@ -1,3 +1,4 @@
+<# Use JSP/eRuby like tagging as curly braces are frequent in LaTeX -#>
 %% Generated by Sphinx.
 \def\sphinxdocclass{<%= docclass %>}
 <% if latex_engine == 'lualatex' -%>
```
or a better one in this and the other latex templates.